libqtile/backend/x11/core.py

```diff
@@ -73,6 +74,39 @@ class ExistingWMException(Exception):
     pass
 
 
+class SelectionStatus(enum.Enum):
+    empty = enum.auto()
+    targets = enum.auto()
+    awaiting = enum.auto()
+    incremental_transfer = enum.auto()
+    available = enum.auto()
+
+
+class Selection:
+    def __init__(self):
+        self.owner = None
+        self.binary_selection = b""
+        self.selection = ""
+        self.status = SelectionStatus.empty
+        self.targets = []
+        self._cur_tar = -1
+
+    @property
+    def current_target(self):
+        return self.targets[self._cur_tar] if self.targets and self._cur_tar >= 0 else "UTF8_STRING"
+
+    # For backwards compatibility
+    def __index__(self, idx):
+        return self.__dict__[idx]
```

Comment:
What kind of backwards compatibility is this needed for? Code people may have hooked to the two selection hooks?

Reply:
Yes, that is correct.

Reply:
Alright, perhaps we should log a deprecation warning so that we can remove this in the future.